invoicer_project/static/js/items_list_request.js

```diff
@@ -0,0 +1,124 @@
+const host = "http://127.0.0.1:8000";
+
+async function getUserData() {
+    let jsonResponse, response;
+    try {
+        const result = await fetch(host + '/items/items_list/', {
+            method: "GET",
+            headers: {
+                'Authorization': `Bearer ${window.localStorage.getItem('accessToken')}`
+            },
+        });
+        response = result.status;
+        jsonResponse = await result.json();
+    } catch (error) {
+        console.error('Going to obtain new access token!');
+    }
+    return {'responseStatus': response, 'data': jsonResponse};
+}
+
+function fillInitials(userData) {
+    const userFirstName = userData["first_name"];
+    const userLastName = userData["last_name"];
+    document.getElementById("user_name").textContent = userFirstName + " " + userLastName;
+}
+
+function createItemListContent(data){
+     for (let i = 0; i < data.length; i++) {
+        let itemName = data[i]['name'];
+        let priceAndCurrency = data[i]['price'] + " " + data[i]['currency'];
+
+        document.getElementById("items_container").insertAdjacentHTML('afterbegin', `<div class="row client_list_item align-items-center justify-content-around">
+                        <div class="col-md-6 col-sm-6 col-7 list_item_name">
+                            <p class="item_name">${itemName}</p>
+                        </div>
+                        <div class="d-flex flex-wrap flex-row justify-content-end col-md-6 col-sm-6 col-5">
+                            <div class="d-flex flex-wrap flex-column list_item_info_block">
+                                <p class="additional_text">Price per unit</p>
+                                <p class="main_text">${priceAndCurrency}</p>
+                            </div>
+                            <div class="d-flex flex-wrap flex-column list_item_info_block">
+                                <p class="additional_text">Basic unit</p>
+                                <p class="main_text">kg</p>
+                            </div>
+                            <div class="list_item_user_buttons">
+                                <span class="material-symbols-outlined" style="font-size: 28px">delete</span>
+                                <md-checkbox id="list_item_user_delete"></md-checkbox>
+                            </div>
+                        </div>
+                    </div>`)
+    }
+}
+
+async function obtainNewAccessToken() {
```

Comment:
If I am not mistaken, this function has been already implemented in another django project? If so, let's reuse it instead of duplication

Reply:
We were thinking about refactoring the code. We are going to fix this.